lib/puppet/file_system.rb

```diff
@@ -32,6 +32,15 @@ def self.overlay(*files, &block)
 
   # Opens the given path with given mode, and options and optionally yields it to the given block.
   #
+  # @param path [String, Pathname] the path to the file to operate on
+  # @param mode [Integer] The mode to apply to the file if it is created
+  # @param options [String] Extra file operation mode information to use
+  #   This is the standard mechanism Ruby uses in the IO class, and therefore
+  #   encoding may be explicitly like fmode : encoding or fmode : "BOM|UTF-*"
```

Comment:
Was this intended to read something like `...may be _supplied_ explicitly like...` ?